File: contests/abc057/abc057_c.cpp
// C - Digits in Multiplication
// https://atcoder.jp/contests/abc057/tasks/abc057_c

#include <iostream>
#include <cmath>
#define ll long long int
using namespace std;

int main() {
    ll n, ans; cin >> n;
    for(int i=1;i<=sqrt(n);i++) {
        if (n%i == 0) {
            ans = min(ans, (ll)to_string(n/i).size());
        }
    }
    cout << ans << endl;
}

File: contests/abc057/abc057_c.py
# C - Digits in Multiplication
# https://atcoder.jp/contests/abc057/tasks/abc057_c

n = int(input())

for i in range(1, int(n**0.5) + 1):
	if n%i == 0:
		ans = len(str(n // i))

print(ans)
